### PR TITLE
#4888 Fix app crashes on tapping save changes button multiple times

### LINKED ISSUE #4888: App crashes on tapping Save changes button multiple times
## Describe the bug
App crashes when we edit some extra info and tap `Save changes` button multiple times and click another button afterwards (`Edit` button now)

### To reproduce

Steps to reproduce the behaviour:
1. IN Vaccinations, create new patient and add complete vaccination event
2. GO back to that patient vaccination history
3. Edit some extra information (like add/edit ID number)
4. Tap `Save Changes` button of Page 2 multiple times. Do same for Extra info page's `Save Changes` button as well.
5. Now click on `Edit` button (of Page 3)
6. See error: app crashes!
7. Re-open and try to click that patient history: app crashes every single time.

### Expected behaviour
No app crashes!

### Proposed Solution
Leave if you don't know how to fix/implement. Edit this issue description and explain here if you know the best path of implementing the fix within the codebase.

### Version and device info
- App version: latest

### PR DESCRIPTION
**Pull request by sworup**
Fixes #4888

## Change summary

Fixes app crashes on tapping save changes button multiple times

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] Go to Vaccinations > Select a Patient's `History`
- [ ] Select a vaccine event from the list to edit
- [ ] After the vaccination event details window opens. Tap on save changes button however you like, tap multiple times in any order you want. 
- [ ] Originally the bug originated from Page 2 Additional Details, just saving it without updating any data would have crashed the app. So check that with priority.
- [ ] The data would be saved, or not saved at all after our current fix. Go back to the Patient list page and navigate back to the same vaccine event. Originally trying to do this would have crashed the app. Now it should work properly.
- [ ] Regression testing: Maybe try a vaccine event from create just to make sure create is working properly.

Out of stock vaccine causes dropdown error
- [ ] Run a vaccine out of stock, preferably with name that would have shown up in the dropdown list in between, this is an index bug afterall. If there are 3 vaccines [a, b, c] and b is out of stock. It would show in the vaccine dropdown as [a, c]. So here c is index 1. But when the selected vaccine is set it set as index 1 of earlier vaccines list which is b. 
- [ ] Go to Vaccinations > Select a Patient's `History` > Select a vaccine event from the list to edit 
- [ ] Edit Vaccination events. 
- [ ] Select a vaccine, preferably one coming after out of stock vaccine in the original list. 
- [ ] Make sure correct vaccine is selected in the dropdown and after save it is saved correctly too.

Cannot save edit of Page 3: Vaccination event
- [ ] Go to Vaccinations > Select a Patient's `History` > Select a vaccine event from the list to edit 
- [ ] Edit Vaccination event.
- [ ] Change vaccinator, change vaccine
- [ ] click Save changes
- [ ] Data should update properly. 

### Related areas to think about

If there are any general areas of the codebase your changes might have side affects on, mention them here

### CODE REVIEW

**Review by chastichno — CHANGES_REQUESTED** [review]
Right after I select the vaccination event in the patient history, it shows me that there are no json forms for the vaccination event, but it disappears after I click once anywhere:

https://user-images.githubusercontent.com/41636693/185837464-4f87f14c-4a19-48cf-8952-f2326f10ca98.mov

I expect it to work as before - forms should load right away after selecting a vaccination event. With this requesting changes

**Comment by sworup** [comment]
> Right after I select the vaccination event in the patient history, it shows me that there are no json forms for the vaccination event, but it disappears after I click once anywhere:

@chastichno I moved the variables inside state. You do not need to click again to trigger form load, this was the component on loading state. I guess this is annoying enough that it needs fix. I will work on it. Does rest of the logic look good???

**Comment by chastichno** [comment]
@sworup Thank you Sworup. Yes, the rest looks good (except for the buttons on which I commented)

File: src/widgets/modals/VaccinationEvent.js [hunk 1]
```diff
@@ -283,13 +372,15 @@ export const VaccinationEventComponent = ({
       </FlexRow>
       <FlexRow flex={0} justifyContent="center">
         <PageButton
+          disabled={!(!!surveySchema && !!surveyForm)}
```
[the remark this answers is not in the document]
Comment (sworup) [reply src/widgets/modals/VaccinationEvent.js]: Resolved with [e82adf2](https://github.com/openmsupply/mobile/pull/4899/commits/e82adf2bd3cdff1418a5f5b1e6e50a70cf1f1cfc)

**Comment by sworup** [comment]
> I still have the same problem with loading the forms. I think it is not a good idea to show users the "No form found" page while it's loading as it looks quite alarming. Can we use a "loading" page instead?

@chastichno resolved with [e2580eb](https://github.com/openmsupply/mobile/pull/4899/commits/e2580eb974fd1496454ccadae285fdaf55acc34b)

**Review by chastichno — APPROVED** [review]
Looks great! Needs to be properly tested though

**Pull request merged**